modin/core/storage_formats/pandas/parsers.py

```diff
@@ -686,15 +672,24 @@ def parse(fname, **kwargs):
 storage_options : dict
     Parameters for specific storage engine.""",
     )
-    def parse_fsspec_files(files_for_parser, storage_options, **kwargs):
+    def parse(files_for_parser, storage_options, **kwargs):
         from pyarrow.parquet import ParquetFile
 
         columns = kwargs.get("columns", None)
 
         chunks = []
+        # `single_worker_read` just passes in a string path
+        if isinstance(files_for_parser, str):
+            return pandas.read_parquet(files_for_parser, **kwargs)
+
         for file_for_parser in files_for_parser:
             if isinstance(file_for_parser.path, IOBase):
-                context = nullcontext(file_for_parser.path)
+                if sys.version_info < (3, 7):
+                    context = NullContextManager(file_for_parser.path)
+                else:
+                    from contextlib import nullcontext
```

Comment:
if we import `NullContextManager` regardless of Python version above, why can't we use it regardless of the version?

Reply:
This is probably the way to go. I didn't know we could do something as elegant as you suggested before. My solution was too verbose and I wasn't sure about its correctness.